dm/dm/master/scheduler/scheduler.go

```diff
@@ -1528,15 +1465,19 @@ func (s *Scheduler) StopRelay(source string, workers []string) error {
 			continue
 		}
 
-		startedRelay := worker.RelaySourceID()
-		if startedRelay == "" {
+		relaySources := worker.RelaySources()
+		if len(relaySources) == 0 {
```

Comment:
I think here should be `_, ok := relaySources[source]` which means the worker has stopped the relay for this source (it can has other relay working), rather than we treat it as `unmatchedWorkers`. `unmatchedWorkers` will return error.

Reply:
Do you mean we should delete this judge and treat this worker as alreadyStopped?
```go
		if _, ok := relaySources[source]; !ok {
```

Reply:
> Do you mean we should delete this judge and treat this worker as alreadyStopped?
> 
> ```go
> 		if _, ok := relaySources[source]; !ok {
> ```

yes